File: ios/LivenessDetection/LivenessError.swift
import Foundation

/// Error codes surfaced via `LivenessDetectorDelegate.onFailure`.
@objc public final class LivenessErrorCodes: NSObject {
  /// CDN/asset host unreachable after retries (internet confirmed).
  @objc public static let cdnNotAvailable = "cdnNotAvailable"

  /// No internet connection.
  @objc public static let offline = "offline"

  @objc public static func isCdnNotAvailable(_ reason: String) -> Bool {
    return reason == cdnNotAvailable
  }

  @objc public static func isOffline(_ reason: String) -> Bool {
    return reason == offline
  }

  private override init() { super.init() }
}

public struct LivenessError: Error {
  public let code: String
  public let message: String

  public init(code: String, message: String) {
    self.code = code
    self.message = message
  }
}

File: android/liveness/src/main/java/com/liveness/detection/LivenessError.kt
package com.liveness.detection

/** Error codes surfaced via [LivenessListener.onFailure]. */
object LivenessErrorCodes {
  /** CDN/asset host unreachable after retries (internet confirmed). */
  const val CDN_NOT_AVAILABLE = "cdnNotAvailable"

  /** No internet connection. */
  const val OFFLINE = "offline"

  fun isCdnNotAvailable(reason: String): Boolean = reason == CDN_NOT_AVAILABLE
  fun isOffline(reason: String): Boolean = reason == OFFLINE
}

class LivenessError(
  val code: String,
  message: String,
) : RuntimeException(message)
